examples/ct/structures_test.go

```diff
@@ -64,31 +63,25 @@ func TestSerializeLogEntry(t *testing.T) {
 	ts := ct.TimestampedEntry{
 		Timestamp:  12345,
 		EntryType:  ct.X509LogEntryType,
-		X509Entry:  ct.ASN1Cert([]byte{0x10, 0x11, 0x12, 0x13, 0x20, 0x21, 0x22, 0x23}),
+		X509Entry:  &ct.ASN1Cert{Data: []byte{0x10, 0x11, 0x12, 0x13, 0x20, 0x21, 0x22, 0x23}},
 		Extensions: ct.CTExtensions{}}
-	leaf := ct.MerkleTreeLeaf{LeafType: ct.TimestampedEntryLeafType, Version: ct.V1, TimestampedEntry: ts}
+	leaf := ct.MerkleTreeLeaf{LeafType: ct.TimestampedEntryLeafType, Version: ct.V1, TimestampedEntry: &ts}
 
 	for chainLength := 1; chainLength < 10; chainLength++ {
 		chain := createCertChain(chainLength)
 
-		var buff bytes.Buffer
-		w := bufio.NewWriter(&buff)
-
 		logEntry := LogEntry{Leaf: leaf, Chain: chain}
-		err := logEntry.Serialize(w)
-
+		entryData, err := tls.Marshal(logEntry)
 		if err != nil {
 			t.Fatalf("failed to serialize log entry: %v", err)
 		}
 
-		w.Flush()
-		r := bufio.NewReader(&buff)
-
 		var logEntry2 LogEntry
-		err = logEntry2.Deserialize(r)
-
+		rest, err := tls.Unmarshal(entryData, &logEntry2)
 		if err != nil {
 			t.Fatalf("failed to deserialize log entry: %v", err)
+		} else if len(rest) > 0 {
+			t.Fatalf("trailing data after serialized log entry")
```

Comment:
This could be an `Errorf`, then you'd see whether the data was just borked or correct-but-with-trailing-stuff as the test continued below